openfoodfoundation/openfoodnetwork
# Separate admin and Spree routes from main file

WARNING: This could break other pull requests that change `config/routes.rb`.

#### What? Why?

In #2512, Code Climate has started [warning us](https://codeclimate.com/github/openfoodfoundation/openfoodnetwork/pull/2512) about the length of `config/routes.rb`.

To make routes management more manageable, I separated the admin and Spree engine routes and moved them to `config/routes/`.

#### What should we test?

Testing that the routes are loaded correctly should already be covered by feature specs.

But this potentially messes up reloading of routes in `development`. To check, I did the following: (Note: Only developers would be able to test this.)

1. Add logging to the start of each route file, for example:
    ```ruby
    Rails.logger.info "-- Loading config/routes.rb"
    ```
2. Load home page to load all route files. You should see the following in the log:
    ```ruby
    -- Loading config/routes.rb
    -- Loading config/routes/admin.rb
    -- Loading config/routes/spree.rb
    ```
3. In one of the three route files, add a route, for example:
    ```ruby
    get :testmain, to: "home#index"
    ```
4. Test the newly added route, e.g. `/testmain`. This should load the assigned page/action.
5. Check the log. The application should have reloaded all route files in the correct order, so you should be able to see the following again:
    ```ruby
    -- Loading config/routes.rb
    -- Loading config/routes/admin.rb
    -- Loading config/routes/spree.rb
    ```
6. Repeat Items 3, 4, and 5 for the two other route files.

#### Release notes

- Split routes into three separate route files.

Changelog Category: Changed

#### How is this related to the Spree upgrade?

If the Spree upgrade includes a major upgrade of Rails, there might be a need to update the way that the other route files are loaded.
